Automattic/simplenote-ios
# Dark Mode: Mark I

### Details:
In this PR we're implementing:

- **UIColorName:** Objective C friendly enum, in charge of representing all of the colors we've got
- **UIColor Extension:** Acts as an adapter, and lets us convert UIColorName(s) into either iOS 13 Dynamic Colors, or VSTheme-procured-colors
- **Bonus:** Some nullability specifiers here and there!

Standing on top of this, we've nuked few explicit color getters we've recently implemented as UIColor extension methods. 

**Every** UIColor instantiation is to be channeled, from now on, thru this new API: `[UIColor colorWithName:]`

Ref. #367 

### Note:
This is a multi stage approach. Foundational baby steps in this PR!

### Testing: Light Mode
1. Launch Simplenote in Light Mode
2. Open the Note List
3. Swipe over any note in the Note List
4. Verify the three actions look correctly
5. Toggle Dark Mode
6. Repeat (3) and verify the actions look well in Dark Mode, as well

### Testing: Share Extension
1. Launch Safari
2. Trigger the Simplenote Share Extension
3. Verify the navigation bar looks as expected
